optimum/onnxruntime/modeling_ort.py

```diff
@@ -308,16 +308,19 @@ def to(self, device: Union[torch.device, str, int]):
         if device.type == "cuda" and self.providers[0] == "TensorrtExecutionProvider":
             return self
 
-        if device.type == "cuda" and self._use_io_binding is False:
+        self.device = device
+        provider = get_provider_for_device(self.device)
+        validate_provider_availability(provider)  # raise error if the provider is not available
+
+        if device.type == "cuda" and self._use_io_binding is False and provider != "ROCMExecutionProvider":
```

Comment:
Is IO Binding not supported with ROCM EP? If so, we should probably mention it in the doc, or in an explicit comment here (maybe with a reference if there is).